instrumentation/opentelemetry-instrumentation-urllib3/src/opentelemetry/instrumentation/urllib3/__init__.py

```diff
@@ -175,11 +203,38 @@ def instrumented_urlopen(wrapped, instance, args, kwargs):
             inject(headers)
 
             with _suppress_further_instrumentation():
+                start_time = default_timer()
                 response = wrapped(*args, **kwargs)
+                elapsed_time = (default_timer() - start_time) * 1000
 
             _apply_response(span, response)
             if callable(response_hook):
                 response_hook(span, instance, response)
+
+            metric_labels = {
+                SpanAttributes.HTTP_METHOD: method,
+                SpanAttributes.HTTP_HOST: instance.host,
+                SpanAttributes.HTTP_SCHEME: instance.scheme,
+                SpanAttributes.HTTP_STATUS_CODE: response.status,
+                SpanAttributes.NET_PEER_NAME: instance.host,
+                SpanAttributes.NET_PEER_PORT: instance.port,
+            }
+
+            version = getattr(response, "version")
+            if version:
+                metric_labels[SpanAttributes.HTTP_FLAVOR] = (
+                    "1.1" if version == 11 else "1.0"
+                )
+
+            request_size = 0 if body is None else len(body)
```

Comment:
Should you be using `Content-Length` from the response headers? Is all body data read in-memory? What if the response is compressed?

Reply:
Good point, I Changed only the response to get the Content-Length, in the request I cant get the Content-Length from the header